tests/go.mod

```
@@ -3,6 +3,7 @@ module github.com/nginxinc/nginx-gateway-fabric/tests
 go 1.22.2
 
 require (
+	github.com/nginxinc/nginx-gateway-fabric v0.1.0-rc.1.0.20240521170716-206f1364bf89
```

```suggestion
	github.com/nginxinc/nginx-gateway-fabric v0.0.0
```
I don't think we want to pin a version here. Then we should add a replace

```
replace github.com/nginxinc/nginx-gateway-fabric => ../
```